### PR TITLE
Added script to CI to delete old nightly built conda packages

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
Added script to CI to delete old nightly built conda packages to have enough space for new packages.

## How Has This Been Tested
Using CI and check if packages got deleted.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: conda-build/common/delete_old_night_packages.py
Comment (erikwijmans) [inline conda-build/common/delete_old_night_packages.py:1]: Hmm, looks the pre-commit hooks aren't enabled on this directory!  Can you run black on this manually for now?

**Review by erikwijmans — APPROVED** [review]
Seems fine.  This likely isn't worth over engineering, so if it works it works!

File: conda-build/common/delete_old_night_packages.py
Comment (aclegg3) [inline conda-build/common/delete_old_night_packages.py:7]: Description is copied?
```suggestion
parser = argparse.ArgumentParser("Deleting old Conda packages.")
```

File: conda-build/common/delete_old_night_packages.py
Comment (aclegg3) [inline conda-build/common/delete_old_night_packages.py:10]:
```suggestion
parser.add_argument('--nightly', help="Remove old conda nightly builds.", action='store_true')
```

**Review by aclegg3 — APPROVED** [review]
Thanks for automating this! 👍

**Review by Skylion007 — APPROVED** [review]
Please run the linters like black on this. Otherwise looks good.

**Comment by Skylion007** [comment]
@mathfac this is blocked by #1087

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Wait until #1087 is merged, then pull from master and run.

File: conda-build/common/delete_old_night_packages.py
Comment (Skylion007) [reply conda-build/common/delete_old_night_packages.py:1]: @erikwijmans Fixed now in another PR. Was a typo in the config.

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
@mathfac Seems like there is a typo in the Python AST?

**Pull request merged**
